packages/drivers/routerlicious-driver/src/documentServiceFactory.ts

```diff
@@ -98,27 +98,57 @@ export class RouterliciousDocumentServiceFactory implements IDocumentServiceFact
         const rateLimiter = new RateLimiter(this.driverPolicies.maxConcurrentOrdererRequests);
-        const ordererRestWrapper = await RouterliciousOrdererRestWrapper.load(
-            tenantId,
-            undefined,
-            this.tokenProvider,
+
+        const ordererRestWrapper = await PerformanceEvent.timedExecAsync(
             logger2,
-            rateLimiter,
-            this.driverPolicies.enableRestLess,
-            resolvedUrl.endpoints.ordererUrl,
-        );
+            {
+                eventName: "GetTokenForCreateNew",
+                details: JSON.stringify({
+                    enableRestLess: this.driverPolicies.enableRestLess,
+                }),
+
+            },
+            async () => {
+                return RouterliciousOrdererRestWrapper.load(
+                    tenantId,
+                    undefined,
+                    this.tokenProvider,
+                    logger2,
+                    rateLimiter,
+                    this.driverPolicies.enableRestLess,
+                    resolvedUrl.endpoints.ordererUrl,
+                );
+            });
 
-        // @TODO: Remove returned "string" type when removing back-compat code
-        const res = await ordererRestWrapper.post<{ id: string; token?: string; session?: ISession; } | string>(
-            `/documents/${tenantId}`,
+        const res = await PerformanceEvent.timedExecAsync(
+            logger2,
             {
-                summary: convertSummaryToCreateNewSummary(appSummary),
-                sequenceNumber: documentAttributes.sequenceNumber,
-                values: quorumValues,
-                enableDiscovery: this.driverPolicies.enableDiscovery,
-                generateToken: this.tokenProvider.documentPostCreateCallback !== undefined,
+                eventName: "CreateNew",
+                details: JSON.stringify({
+                    enableDiscovery: this.driverPolicies.enableDiscovery,
+                    sequenceNumber: documentAttributes.sequenceNumber,
+                }),
             },
-        );
+            async (event) => {
+                // @TODO: Remove returned "string" type when removing back-compat code
+                const postRes = await ordererRestWrapper.post<
+                { id: string; token?: string; session?: ISession; } | string
+                >(
+                    `/documents/${tenantId}`,
+                    {
+                        summary: convertSummaryToCreateNewSummary(appSummary),
+                        sequenceNumber: documentAttributes.sequenceNumber,
+                        values: quorumValues,
+                        enableDiscovery: this.driverPolicies.enableDiscovery,
+                        generateToken: this.tokenProvider.documentPostCreateCallback !== undefined,
+                    },
+                );
+
+                event.end({
+                    docId: (typeof postRes === "string") ? postRes : postRes.token,
```

Comment:
postRes.token will return a JWT. I think you're looking for postRes.id

Reply:
Ahh good find